sio/executors/checker.py

```diff
@@ -129,9 +130,42 @@ def run(environ, use_sandboxes=True):
         environ['result_code'] = 'OK'
         if output[1]:
             environ['result_string'] = _limit_length(output[1])
-        environ['result_percentage'] = float(output[2] or 100)
+        if isinstance(output[2], bytes):
+            output[2] = output[2].decode('utf-8')
+        environ['result_percentage'] = output_to_fraction(output[2].strip())
+        if environ['result_percentage'][1] == 0:
+            raise CheckerError(
+                'Checker returned 0 as denominator. Checker stdout: "%s". Checker environ dump: %s'
+                % (output[2], environ)
+            )
     else:
         environ['result_code'] = 'WA'
         environ['result_string'] = _limit_length(output[1])
-        environ['result_percentage'] = 0
+        environ['result_percentage'] = (0, 1)
     return environ
+
+
+def output_to_fraction(output_str):
+    if not output_str:
+        return 100, 1
+    output_is_float = re.match(r"^[0-9]+\.[0-9]*$", output_str)
+    output_is_percent = re.match(r"^[0-9]+$", output_str)
+    output_is_fraction = re.match(r"^[0-9]+ [0-9]+$", output_str)
+    if output_is_float:
+        return float_to_fraction(output_str)
```

Comment:
it can be simplified with:
```
Decimal(output_str).as_integer_ratio()
```

Reply:
actually it can be simplified even more. python class Fraction supports formats:
* A/B Fraction("42/123").as_integer_ratio()
* A (like Fraction("42").as_integer_ratio()
* A.B (like Fraction("42.123").as_integer_ratio())